spec/components/playbacktimelabel.spec.ts

```diff
@@ -73,8 +78,30 @@ describe('PlaybackTimeLabel', () => {
 
       jest.spyOn(playerMock, 'getDuration').mockReturnValue(100);
       playbackTimeLabel.configure(playerMock, uiInstanceManagerMock);
+      playerMock.eventEmitter.fireEvent<PlayerEventBase>(
+        {type: PlayerEvent.Ready, timestamp: Date.now()}
+      );
       expect(playbackTimeLabel.getText()).toEqual('01:40');
     });
+
+    it('displays the total time hh:mm:ss', () => {
+      jest.spyOn(playerMock, 'isLive').mockReturnValue(false);
+      playbackTimeLabel = new PlaybackTimeLabel({ timeLabelMode: PlaybackTimeLabelMode.TotalTime });
+
+      // Setup DOM Mock
+      const mockDomElement = MockHelper.generateDOMMock();
+      jest.spyOn(playbackTimeLabel, 'getDomElement').mockReturnValue(mockDomElement);
+
+      jest.spyOn(playerMock, 'getDuration').mockReturnValue(0);
+      expect(playbackTimeLabel.getText()).toEqual(undefined);
+
+      jest.spyOn(playerMock, 'getDuration').mockReturnValue(3600);
```

Comment:
I would create a separate test for this where you explicitly test the switch from 0 to something greater than 1 hour based on the given event.

IMO to really test the use-case we are talking about you would need to emit a SourceLoaded event between changing the duration

Reply:
at the end we used `Ready`